m2cgen/interpreters/interpreter.py

```diff
@@ -203,9 +230,15 @@ def interpret_tanh_expr(self, expr, **kwargs):
         nested_result = self._do_interpret(expr.expr, **kwargs)
         return self._cg.function_invocation(self.tanh_function_name, nested_result)
 
-    def _wrap_infix_expr(self, expr, left_precedence, right_precedence):
+    def _wrap_infix_expr(self, expr, left_precedence, right_precedence, right_is_associative):
         wrap = left_precedence is not None and left_precedence > expr.precedence
-        wrap |= right_precedence is not None and right_precedence >= expr.precedence
+        wrap |= (
+            right_precedence is not None
+            and (right_precedence > expr.precedence
+                 or (right_precedence == expr.precedence
+                     and right_is_associative is not None
```

Comment:
[Nitpick] can we reduce the condition to just `right_is_associative == False`? I realize it's a little counterintuitive.

Reply:
Ah, indeed, thanks a lot for the tip!
But we should use `is False` instead of `== False` according to the
https://www.flake8rules.com/rules/E712.html